openfe/protocols/openmm_afe/afe_settings.py

```diff
@@ -0,0 +1,386 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+"""Settings class for equilibrium AFE Protocols using OpenMM + OpenMMTools
+
+This module implements the necessary settings necessary to run absolute free
+energies using :class:`openfe.protocols.openmm_abfe.equil_abfe_methods.py`
+
+
+TODO
+----
+* Add support for restraints
+* Improve this docstring by adding an example use case.
+
+"""
+from typing import Optional
+from pydantic import validator
+from openff.units import unit
+from gufe import settings
+
+
+class SystemSettings(settings.SettingsBaseModel):
+    """Settings describing the simulation system settings."""
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    nonbonded_method = 'PME'
+    """
+    Method for treating nonbonded interactions, currently only PME and
+    NoCutoff are allowed. Default PME.
+    """
+    nonbonded_cutoff = 1.0 * unit.nanometer
+    """
+    Cutoff value for short range nonbonded interactions.
+    Default 1.0 * unit.nanometer.
+    """
+
+    @validator('nonbonded_method')
+    def allowed_nonbonded(cls, v):
+        if v.lower() not in ['pme', 'nocutoff']:
+            errmsg = ("Only PME and NoCutoff are allowed nonbonded_methods")
+            raise ValueError(errmsg)
+        return v
+
+    @validator('nonbonded_cutoff')
+    def is_positive_distance(cls, v):
+        # these are time units, not simulation steps
+        if not v.is_compatible_with(unit.nanometer):
+            raise ValueError("nonbonded_cutoff must be in distance units "
+                             "(i.e. nanometers)")
+        if v < 0:
+            errmsg = "nonbonded_cutoff must be a positive value"
+            raise ValueError(errmsg)
+        return v
+
+
+class SolventSettings(settings.SettingsBaseModel):
+    """Settings for solvating the system
+
+    NOTE
+    ----
+    * No solvation will happen if a SolventComponent is not passed.
+    """
+    solvent_model = 'tip3p'
+    """
+    Force field water model to use.
+    Allowed values are; `tip3p`, `spce`, `tip4pew`, and `tip5p`.
+    """
```

Comment:
Out of curiosity, where does this load parameters from? OpenMM?

Reply:
This is the solvent model used for addSolvent, which I believe is using XML files baked into OpenMM. We then override the FF using whatever user provided XML in the SystemGenerator settings (in practice it has to be from openmmforcefields).